tools/depmod.c

```diff
@@ -2306,32 +2304,37 @@ static int output_symbols_bin(struct depmod *depmod, FILE *out)
 
 	hash_iter_init(depmod->symbols, &iter);
 
+	strbuf_init_with_stack(&salias, 1024);
+	strbuf_pushmem(&salias, base, baselen);
+
 	while (hash_iter_next(&iter, NULL, &v)) {
 		int duplicate;
 		const struct symbol *sym = v;
-		size_t len;
-		char *s;
+		size_t sym_sz;
 
 		if (sym->owner == NULL)
 			continue;
 
-		len = strlen(sym->name);
+		sym_sz = strlen(sym->name) + 1;
 
-		if (scratchbuf_alloc(&salias, baselen + len + 1) < 0) {
+		if (!strbuf_reserve(&salias, sym_sz)) {
```

Comment:
Why did we drop `baselen` from the total/reservation size?

Reply:
scratchbuf_alloc() passes the total space. strbuf_reserve() reserves **adiditional** space... here we don't need to track the total space since we add popchars/popchar where needed. On a hindsight just _reserve() may not be clear enough about what's the behavior. std::vector::reserve() is actually about the total sapce. Linux kernel's xarray is about 1 position.

Maybe add 2 functions?
strbuf_reserve() / strbuf_reserve_total()
strbuf_alloc() / strbuf_alloc_total()
strbuf_ensure_space() / strbuf_ensure_total_space()
strbuf_add() /

Reply:
Indeed C++ came to mind when I read `strbuf_reserve()` and didn't think too much about it :facepalm: The xarray one never crossed my mind, since it takes an instance/index and not a size as an argument.

Having two functions makes sense. Alternatively we can use something like `strbuf_reserve_extra()` to indicate the "in-addition to" aspect. Don't have any strong preference fwiw.

Reply:
strbuf_reserve_extra() looks like a good name. I will change to have just that one